Bug BB-8 and BB-9 was fixad. ApplicationError exceptions was refactored

ivr42 opened the pull request:
- `/src/bot/main.py` now handles all unhandled types with an error message
 - `photo_handler` in `/src/bot/handlers.py` correctly handle exceptions now
  - All exceptions are now inherited from `fastapi.HTTPException` class
  - All exceptions now have the suffix `Error`
  - In all modules, exceptions are imported as
    `from src.core import exceptions` and used as `exceptions.<ExceptionClass>`

Notion tasks:
 - [ВВ-8](https://www.notion.so/8-ab12bb2c1f6140be9ab3c13913088230)
 - [ВВ-9](https://www.notion.so/9-ea68f7dac28e4224a25f9b0139a8b94a)
 - [Bot: рефакторинг метода photo_handler](https://www.notion.so/Bot-photo_handler-aed00c40163a4cc08a0d9b7a7ef7046e)

ivr42 committed:
Bug BB-8 and BB-9 was fixad. ApplicationError exceptions was refactored

 - `/src/bot/main.py` now handle all unhendled types with error message
 - `photo_handler` in `/src/bot/handlers.py` correctly handle exceptions now
 - ApplicationError exceptions was refactored

AntonZelinsky commented on src/core/exceptions.py:
```diff
@@ -24,67 +24,85 @@ def __init__(self, detail: str | None) -> None:
+        super().__init__()
+
+
+class ApplicationException(HTTPException):
```
А в чём разница межу `ApplicationError` и `ApplicationException`? И зачем на две сущости? Мне кажется, это будет только путать всех и не будет понятно, когда и что использовать.
Предлагаю оставить один, только определиться в стандарте именования, ошибка это или всё же исключение.

ivr42 replied:
Ок, я в принципе согласен. Я действительно не понимаю зачем и почему было сделано два отдельных родительских класса (их было больше — там было много классов ошибок, которые наследовались напрямую от `Exception`).
Возможно тут была какая-то идея. 
Я сделал это в качестве предложения по оптимизации — в `/src/bot/handlers.py` из-за этого было много дублирования кода. 

Я посмотрю сегодня есть ли, и если есть то какие подводные камни объединения всех исключений под один родительский класс.

ADesBiysk replied:
К сожалению, с исключениями сейчас бардак. Пытались навести порядок в этом ПР https://github.com/Studio-Yandex-Practicum/lomaya_baryery_backend/pull/192, но это требовало очень больших изменений во всех файлах, поэтому отложили. 
2 сущности появилось из-за того, что нужно было в одной из задач, как и здесь, вынести одно исключение, а все остальные отнаследовали от него. Это бы потребовало большого количества правок, поэтому "временно" :) добавили второе исключение. 
По поводу имени. Если брать стандартные исключения, то там используется Error. 
![python-exception-heirarchi](https://user-images.githubusercontent.com/26863439/225426431-00eff1a4-f07e-4cfe-a1f8-9591e26bc0d2.png)
1. Я предлагаю использовать Error в названии.
2. Импортировать их в файл как from src.core import exceptions, чтобы было понятно какие исключения наши.

ivr42 replied:
Учёл все замечания и поправил

ivr42 committed:
Exceptions were refactored

  - All exceptions are now inherited from `fastapi.HTTPException` class
  - All exceptions are now have suffix `Error`
  - In all modules, exceptions are imported as
    `from src.core import exceptions` and used as `exceptions.<ExceptionClass>`

ivr42 commented:
Делим пул-реквест на 3
1. bugfix BB-8 and BB-9
2. рефакторинг исключений
3. рефакторинг метода photo_handler
